[bug] Conan package option conflicts do not lead to error

### Describe the bug

Environment:
MacOS 14, apple-clang 16, conan 2.8.0

[settings]
arch=x86_64
build_type=Release
compiler=apple-clang
compiler.cppstd=gnu17
compiler.libcxx=libc++
compiler.version=16
os=Macos


Description:

Conan options handling is flawed:

Hard-coded package options may be overridden silently without the user noticing that something bad is happening under the hood.

Conflicting options should lead to an error instead of silently being modified.

Example scenario:
https://github.com/stevn/conan2_options_test


## Introduction

In this reproduction scenario a library package `stuff/1.0.0` is required by an application package `stufftool/1.0.0`.

There is a (transitive) dependency on `libtiff`:

- The `stuff` library package requires libtiff directly.
- The `stufftool` application package requires `opencv`, which in turn requires `libtiff`.
- The application doesn't really know / care that `opencv` and `stuff` each have `libtiff` as dependency. It is agnostic to that transitive dependency, since it doesn't use libtiff itself.

In this example, let's assume that we want to disable the `jbig` option of `libtiff`. The JBIG package is very problematic for closed-source software because it is licensed under GPL.

Therefore the `stuff` library makes sure to disable the `jbig` option by hard-coding the option to `False` in the `configure()` function of its `conanfile.py` recipe.

## Build + Reproduce

On Unix-like operating systems run:

    ./create_all.sh

This currently produces this error:

    CMake Error: Could not create named generator DummyGeneratorToPreventBuildingOfJBIG

Here the build of `jbig` has been made to fail on purpose by setting a non-existent CMake generator name `DummyGeneratorToPreventBuildingOfJBIG` for `jbig` in `global.conf`.
This shall stop the build process as soon as Conan attempts to automatically build JBIG, because we do not want that.

## Error summary

When building `stufftool`, its `opencv` dependency defaults to `libtiff` with **`jbig=True`** option. This leads to its `stuff` dependency silently becoming "corrupted" in a certain way.

Even though it configures a really important option, the **hard-coded option value `jbig=False` is ignored** and the `stuff` library is rebuilt with `jbig=True`! The stuff library with JBIG enabled should never exist, as it is explicitly disabled in its recipe.

Conan apparently automatically tries to "make things work" by only setting the options of a package once and then ignoring the later configure() calls. That seems problematic.

## Expected behavior

I would have expected Conan to **throw an error message** stating that a **package options conflict** has been found in the dependency tree.


### How to reproduce it

Run these commands:

    git clone https://github.com/stevn/conan2_options_test
    cd conan2_options_test
    ./create_all.sh

--- a/conan/cli/printers/graph.py
+++ b/conan/cli/printers/graph.py
@@ -91,6 +91,7 @@ def print_graph_basic(graph):
                     output.info(f"        {src_ref}->{option}={conflict_value}", Color.BRIGHT_CYAN)
         output.info("    It is recommended to define options values in profiles, not in recipes",
                     Color.BRIGHT_CYAN)
+        output.warning("There are options conflicts in the dependency graph", warn_tag="risk")
 
 
 def print_graph_packages(graph):

--- a/test/integration/options/options_test.py
+++ b/test/integration/options/options_test.py
@@ -744,11 +744,16 @@ class TestImportantOptions:
 
 class TestConflictOptionsWarnings:
 
-    def test_options_warnings(self):
+    @pytest.mark.parametrize("important", [True, False])
+    def test_options_warnings(self, important):
         c = TestClient()
         liba = GenConanfile("liba", "0.1").with_option("myoption", [1, 2, 3], default=1)
         libb = GenConanfile("libb", "0.1").with_requires("liba/0.1")
-        libc = GenConanfile("libc", "0.1").with_requirement("liba/0.1", options={"myoption": 2})
+        if important:
+            libc = GenConanfile("libc", "0.1").with_requirement("liba/0.1") \
+                                              .with_default_option("liba*:myoption!", 2)
+        else:
+            libc = GenConanfile("libc", "0.1").with_requirement("liba/0.1", options={"myoption": 2})
         app = GenConanfile().with_requires("libb/0.1", "libc/0.1")
 
         c.save({"liba/conanfile.py": liba,
@@ -767,3 +772,4 @@ class TestConflictOptionsWarnings:
                 It is recommended to define options values in profiles, not in recipes
             """)
         assert expected in c.out
+        assert "WARN: risk: There are options conflicts in the dependency graph" in c.out

